apps/omg_performance/lib/omg_performance/byzantine_events/generators.ex

```diff
@@ -0,0 +1,128 @@
+# Copyright 2019 OmiseGO Pte Ltd
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+defmodule OMG.Performance.ByzantineEvents.Generators do
+  @moduledoc """
+  Provides helper functions to generate spenders for perftest,
+  Streams transactions, utxo positions and blocks using data from Watcher.
+  """
+
+  alias OMG.Eth
+  alias OMG.Eth.RootChain
+  alias OMG.State.Transaction
+  alias OMG.Utxo
+  alias OMG.Watcher.HttpRPC.Client
+
+  require Utxo
+
+  @child_chain_url Application.get_env(:omg_watcher, :child_chain_url)
+
+  @doc """
+  Creates addresses with private keys and funds them with given `initial_funds` on geth.
+  """
+  @spec generate_users(non_neg_integer, [Keyword.t()]) :: [OMG.TestHelper.entity()]
+  def generate_users(size, opts \\ [initial_funds: trunc(:math.pow(10, 18))]) do
+    async_generate_user = fn _ -> Task.async(fn -> generate_user(opts) end) end
+
+    async_generate_users_chunk = fn chunk ->
+      chunk
+      |> Enum.map(async_generate_user)
+      |> Enum.map(&Task.await(&1, :infinity))
+    end
+
+    1..size
+    |> Enum.chunk_every(10)
+    |> Enum.map(async_generate_users_chunk)
+    |> List.flatten()
+  end
+
+  @doc """
+  Streams blocks from child chain rpc starting from the first block.
+  """
+  @spec stream_blocks(child_chain_url: binary()) :: [OMG.Block.t()]
+  def stream_blocks(child_chain_url \\ @child_chain_url) do
+    {:ok, interval} = RootChain.get_child_block_interval()
+
+    Stream.iterate(1, &(&1 + 1))
+    |> Stream.map(&get_block!(&1 * interval, child_chain_url))
```

Comment:
please expand this

Reply:
What exactly you mean by expand here?

Reply:
do not use single pipe statements
`Stream.map(Stream.iterate(1, &(&1 + 1)), &get_block!(&1 * interval, child_chain_url))`